app/models/bookmark.rb

```diff
@@ -0,0 +1,6 @@
+class Bookmark < ActiveRecord::Base
+  belongs_to :topic
+
+  validates :url, presence: true
```

Comment:
do bookmarks have any context if not associated to a topic?  will the app work correctly if they're ever not associated to a topic?  if either is no, then there's probably another validation that would be important.